impl/src/main/java/org/eclipse/parsson/JsonGeneratorKeyImpl.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) 2012, 2021 Oracle and/or its affiliates. All rights reserved.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0, which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception, which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ */
+
+package org.eclipse.parsson;
+
+import jakarta.json.stream.JsonGenerator;
+
+final class JsonGeneratorKeyImpl {
+
+    static JsonGenerator.Key createKey(String name) {
+        return new Key(quoteEscape(name));
+    }
+
+    /**
+     * A key that is already escaped so then can just be written directly.
+     */
+    private static class Key implements JsonGenerator.Key {
+
+        private final char[] escapedName;
+
+        Key(String escapedName) {
+            this.escapedName = escapedName.toCharArray();
+        }
+
+        @Override
+        public char[] toCharArray() {
+            return escapedName;
+        }
+    }
+
+    private static String quoteEscape(CharSequence string) {
+        StringBuilder builder = new StringBuilder();
```

Comment:
Same escape logic as `JsonGeneratorImpl.writeEscapedString()` but writes to the `StringBuilder` instead